build.gradle

```diff
@@ -224,10 +224,51 @@ configurations.all {
         force "io.netty:netty-common:${versions.netty}"
         force "io.netty:netty-handler:${versions.netty}"
         force "io.netty:netty-transport:${versions.netty}"
-        force "io.netty:netty-transport-native-unix-common:${versions.netty}"
+        }
+}
+
+//create source set 'newTest'
+//add classes from the main source set to the compilation and runtime classpaths of the newTest
+sourceSets {
+    newTest {
+        compileClasspath += sourceSets.main.output
+        runtimeClasspath += sourceSets.main.output
     }
 }
 
+//add new task that runs new tests
+task newTest(type: Test) {
+    description = 'Run new tests.'
+    group = 'verification'
+    testClassesDirs = sourceSets.newTest.output.classesDirs
+    classpath = sourceSets.newTest.runtimeClasspath
+
+    //run the newTest task after the test task
+    shouldRunAfter test
+}
+
+//run the newTest task before the check task
+check.dependsOn newTest
```

Comment:
Could we rename these tests to something more like 'integration tests', using the word new will become quickly outdated

Reply:
`integrationTests` would also not be quite precise, as we want to mix integration tests and unit tests. Just doing integration tests would not be feasible if we want a relastic runtime for the tests.

Reply:
Unit tests have different considerations than integration tests - I would imagine them to be authored very differently.  The largest distinction being that unit tests should only be testing the class under test, and not spinning up other components/OpenSearch instances.

Do you have an example of how a unit test would be intermixed in this framework?

Reply:
Of course, the framework will be usually not used for unit tests. However, if the goal of the project is to replace the existing tests, then we need to create new unit tests as well, as it will be unfeasible to get a complete coverage by integration tests.

Reply:
> However, if the goal of the project is to replace the existing tests,

While this is a great vision, it makes execution difficult.  A separate effort for unit tests with appropriate code coverage avoids having to appeal to many different goals at once.  What do you think about changing the goal of this PR to create a way to better maintain and author integration tests, then we can add/replace/rewrite IT?

Reply:
I would agree with @peternied , let's move step by step. Step 1 would be to provide an IT authoring framework. Step n may or may not be to combine IT and non-IT tests. We can see how it goes and decide later.